nvflare/client/model_registry.py

```diff
@@ -52,7 +53,7 @@ def receive(self):
     def get_model(self):
         if not self.cache_loaded:
             self.receive()
-        return self.cached_model
+        return copy.deepcopy(self.cached_model)
```

Comment:
This one is interesting,

IF we don't return a copy of the model, and users in their training code DIRECTLY modify the model params, for example:

```
input_model = flare.receive()
input_model.params = new_weights
```

Then this change will CHANGE our cached original model.

This would cause a problem when we try to calculate the model difference for them.
It would always be all zero (since our cached model's weights have been changed).

What should we do in this situation?
Should we NOT do the copy and just warn the user not to modify this model.params directly?
Ask them to always create a new FLModel object?